service/routes.py

```diff
@@ -97,6 +97,21 @@ def read(promotion_id):
     app.logger.info("Returning promotion: %s", promotion.promotion_name)
     return jsonify(promotion.serialize()), status.HTTP_200_OK
 
+
+@app.route("/promotions/<int:promotion_id>", methods=["DELETE"])
+def delete(promotion_id):
+    """Deletes a Promotion with promotion_id with the fields included in the body of the request"""
+    app.logger.info(f"Got request to delete Promotion with id: {promotion_id}")
+    promotion = Promotion.find(promotion_id)
+    if not promotion:
+        abort_with_error(status.HTTP_404_NOT_FOUND,
+                         f"Promotion with id: {promotion_id} not found")
+    try:
+        promotion.delete()
+        return jsonify({}), status.HTTP_200_OK
+    except Exception as error:  # pylint: disable=broad-except
+        return abort_with_error(status.HTTP_500_INTERNAL_SERVER_ERROR, f"An error occurred : {error}")
```

Comment:
I think we can remove this and leave it to the error handler instead - i updated this in the newest PR you can reference

Reply:
I should delete the test which was created to catch this too right?

Reply:
Done.